ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/metrics/DeltaFilesMetricReporter.java

```diff
@@ -84,6 +84,8 @@
 
   public static final String OBJECT_NAME_PREFIX = "metrics:type=compaction,name=";
 
+  private static long lastSuccessfulLoggingTime = 0;
```

Comment:
This should probably be initialized to System.currentTimeMillis();

Reply:
It is initialized with that value when the `logDeltaDirMetrics()` method is called for the first time.